src/dataviews/histogram-dataview-model.js

```diff
@@ -366,7 +369,19 @@ module.exports = DataviewModelBase.extend({
   },
 
   _onFieldsChanged: function () {
-    if (!helper.hasChangedSomeOf(['offset', 'bins', 'aggregation'], this.changed)) {
+    const start = this.get('start');
+    const end = this.get('end');
+
+    const startEndChanged = helper.hasChangedSomeOf(['start', 'end'], this.changed);
+    const startEndValid = _.isFinite(start) && _.isFinite(end);
+    const hasDifferentValues = this._totals.get('start') !== start || this._totals.get('end') !== end;
+
+    if (startEndChanged && startEndValid && hasDifferentValues) {
+      this._totals.set({ start, end });
+      this._totals.refresh();
```

Comment:
Why the `refresh` call?

I'm worried about the code below, where there are more `set` to `this._totals`. If they really don't have any conflict, it'd be better to encapsulate all the `startEnd` logic in a function. We're mixing here the new behaviour with the existing one. It's hard to read.

Reply:
The refresh call is basically to avoid handling the `change` event in the `data-model` because we would have to duplicate the same logic for the `onFieldsChanged` just to call refresh, so I thought it was easier just to do it here.